### PR TITLE
Version support for cli and binaries

### PR DESCRIPTION
**Pull request by WhiteWalker608**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -17,6 +18,9 @@ func main() {
 		fmt.Println("supervisorctl not installed!!! Please install and try again")
 		return
 	}
+	if util.CheckAndUpdate() { // if update is found then new process with same arguments will be spawned
+		return
+	}
```
Comment (roshanr95) [inline main.go:23]: what's the idea here? that the user would run the same thing again after the update?

File: relay/create.go [hunk 2]
```diff
@@ -64,6 +67,18 @@ func CreateCommand() *cli.Command {
...
+				Usage:       "--version <NUMBER>",
+				Value:       "latest",
+				Destination: &version,
+			},
+			&cli.StringFlag{
+				Name:        "chain-version",
+				Usage:       "--chain-version <NUMBER>",
+				Value:       "latest",
+				Destination: &chain_version,
+			},
```
Comment (roshanr95) [inline relay/create.go:81]: rename chain-version to abci-version

File: util/fetch.go [hunk 3]
```diff
@@ -61,3 +63,31 @@ func Fetch(url, path, usr string, isExecutable bool, overwrite bool) error {
 
 	return nil
 }
+
+func FetchLatestVersion(configName string) (string, error) {
+
+	resp, err := http.Get("https://storage.googleapis.com/marlin-artifacts/bin/config.json")
```
Comment (roshanr95) [inline util/fetch.go:69]: rename to versions.json or something if it has just version key-value pairs

**Review by roshanr95 — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (WhiteWalker608) [reply main.go:23]: No, the user would'nt have to run again. This process will create an updated child process with same arguments and wait for it to complete and then simply return

File: relay/create.go (hunk 2 above)
Comment (WhiteWalker608) [reply relay/create.go:81]: ok

File: util/fetch.go (hunk 3 above)
Comment (WhiteWalker608) [reply util/fetch.go:69]: ok

File: main.go (hunk 1 above)
Comment (roshanr95) [reply main.go:23]: ah, cool

**Review by roshanr95 — APPROVED** [review]

**Pull request merged**